sop-website/docs/Waveform-Data/Waveform-Extension.mdx

```diff
@@ -0,0 +1,222 @@
+| 1     | waveform_registry_id         | int       | Yes         | Generate a unique surrogate key for each waveform file. Use an auto-incremented sequence or UUID logic. Must be persistent across ETL reruns.                                                                                                                                                                                                                                                                                                                                     |
+| 2     | waveform_occurrence_id       | int       | Yes         | Foreign key to the waveform_occurrence table. Must be resolved before file ingestion by matching session ID or aligning timestamps. Raise an exception if missing.                                                                                                                                                                                                                                                                                                                |
+| 3     | waveform_feature_id          | int       | No          | Populate only if the file in this row is itself a feature representation (e.g., a derived high-density time series, spectrogram, or vectorized representation). This value must point to an existing waveform_feature.waveform_feature_id. Use this when a file represents the output of a feature pipeline, not a raw signal. Leave null for traditional raw waveform files. Ensures unambiguous linkage when calculating additional features on derived signal representations. |
+| 4     | person_id                    | int       | Yes         | Inherit directly from the linked waveform_occurrence. Do not independently derive from file metadata. This ensures consistency across all tables.                                                                                                                                                                                                                                                                                                                                 |
+| 5     | waveform_file_start_datetime | datetime  | Yes         | Extract from the file header (e.g., EDF+, WFDB .hea, HDF5 metadata). If unavailable, fallback to waveform_occurrence_start_datetime but log as approximate.                                                                                                                                                                                                                                                                                                                       |
```

Comment:
For WFDB multi-segment recordings, I'm not sure that falling back to `waveform_occurrence_start_datetime` makes sense. In that case the `waveform_file_start_datetime` could be the same for all of the sequential segments.

I don't have a good solution other than perhaps marking it as unknown / NULL.